packages/flutter/lib/src/widgets/nested_scroll_view.dart

```diff
@@ -284,12 +284,85 @@ class NestedScrollView extends StatefulWidget {
   }
 
   @override
-  _NestedScrollViewState createState() => _NestedScrollViewState();
+  NestedScrollViewState createState() => NestedScrollViewState();
 }
 
-class _NestedScrollViewState extends State<NestedScrollView> {
+/// The [State] for a [NestedScrollView].
+///
+/// It allows you to obtain the [ScrollController]s of the children of your [NestedScrollView].
+///
+/// You can access the inner scroll controller via [innerController] and the outer
+/// controller with [outerController], however, notice that the outer scroll position
+/// can also be obtained using [NestedScrollView.controller].
+///
+/// If you want to access the inner scroll controller of your nested scroll view,
+/// you can get its [NestedScrollViewState] by supplying a `GlobalKey<NestedScrollViewState>`
+/// to your [NestedScrollView] widget (the `key` parameter).
+///
+/// {@tool sample}
+///
+/// You would access the [NestedScrollViewState] using a [GlobalKey].
+/// Using the following setup, you can access the inner scroll controller
+/// using `globalKey.currentState.innerController`.
+///
+/// ```dart
+/// class ExampleWidget extends StatelessWidget {
+///   final GlobalKey<NestedScrollViewState> globalKey = GlobalKey();
+///
+///   @override
+///   Widget build(BuildContext context) {
+///     return NestedScrollView(
+///       key: globalKey,
+///       headerSliverBuilder: (context, _) => const [SliverAppBar()],
+///       body: const CustomScrollView(
+///         // ...
+///       ),
+///     );
+///   }
+///
+///   ScrollController get innerController {
+///     return globalKey.currentState.innerController;
+///   }
+/// }
+/// ```
+///
+/// {@end-tool}
+///
+/// Alternatively, you could also visit child elements of your parent widget to obtain
+/// the state if you know exactly that the widget is a direct descendant
+/// (see [BuildContext.visitChildElements] or [Element.visitChildren]).
+class NestedScrollViewState extends State<NestedScrollView> {
   final SliverOverlapAbsorberHandle _absorberHandle = SliverOverlapAbsorberHandle();
 
+  /// The [ScrollController] provided to the [ScrollView] in [NestedScrollView.body].
+  ///
+  /// Manipulating the [ScrollPosition] of this controller fully pushes the header sliver up,
+  /// i.e. the position of the [outerController] will be set to [ScrollPosition.maxScrollExtent],
+  /// unless you use [ScrollPosition.setPixels].
+  /// Visually, the header sliver will be not be visible, i.e. it is "pushed" up out of view.
```

Comment:
It's not clear what you mean by this sentence. If it wants, the header sliver can stay visible (think about a pinned app bar).